src/init.cpp

```diff
@@ -1265,16 +1264,19 @@ bool AppInitMain(NodeContext& node)
         }
     }
 
+    assert(!node.scheduler);
+    node.scheduler = MakeUnique<CScheduler>();
+
     // Start the lightweight task scheduler thread
-    CScheduler::Function serviceLoop = std::bind(&CScheduler::serviceQueue, &scheduler);
+    CScheduler::Function serviceLoop = std::bind(&CScheduler::serviceQueue, node.scheduler.get());
```

Comment:
style-nit in commit 047c1fd23ab7f281af7acd238f3bbb441817686c:

As you are changing this line anyway, might as well get rid of the std::bind mess


```cpp
CScheduler::Function serviceLoop = [&] { node.scheduler->serviceQueue(); };